docs/sdk-configuration.md

```diff
@@ -0,0 +1,676 @@
+
+To allow users of the agent to apply tracing to their own code, the agent should attempt to
+instrument dependency injection to provide an instance of `OpenTelemetry` using the agent-configured
+SDK, for example it should add it to the Spring `ApplicationContext`. For cases where dependency
+injection is not available, though, there is no option but to provide access to the SDK through a
+global variable. We can expect such usage to still function correctly even if the agent is removed
+and a different configuration mechanism is used, such as manual configuration as above, or Spring
+Sleuth.
+
+### SDK Auto-Configuration Wrapper
+
+For non-agent users, we can still provide a non-programmatic solution for configuring the SDK -
+it can be a different artifact which contains SPIs similar to what we have currently, supports
+environment variables and other auto-configuration. A single entrypoint method, `initialize()` could
+determine the configuration, initialize `OpenTelemetry`, and set it as the global. As this artifact
+is in our control, it would be reasonable for `opentelemetry-api` to check the classpath for the
+presence of the wrapper and invoke it automatically.
+
+### Spring Sleuth
+
+[Spring Sleuth](https://spring.io/projects/spring-cloud-sleuth) (or any similar observability-aware server framework such as 
+[curio-server-framework](https://github.com/curioswitch/curiostack/blob/master/common/server/framework/src/main/java/org/curioswitch/common/server/framework/monitoring/MonitoringModule.java)
+or internal frameworks developed by devops teams at companies) is also a mechanism for automatically
+configuring the SDK. In general, we would expect Sleuth users to not be using the java agent.
+
+Examples of how Sleuth could work are presented above in examples using `@Bean`. In particular, we
+expect it to have its own set of configuration properties - by making sure we don't implement
+configuration properties in the core SDK, only configuration layers like the agent or a possible
+configuration wrapper, we avoid the possibility of confusion by having duplicate variables (in
+practice, OpenTelemetry naming would likely be ignored and overwritten by Spring naming).
+
+## Partial SDKs
+
+We allow implementing particular signals of the OpenTelemetry API without using our SDK. For example,
+a MeterProvider may be implemented with micrometer. For this reason, each signal must also present
+all of its options in the form of, e.g., `TracerSdkProviderBuilder`. We expect the vast majority of
+users to use `OpenTelemetrySdkBuilder` - while there is some duplication with the signal provider
+builder, it is work maintainers can do to present the simplest interface for the most common use
+case of using the whole SDK.
+
+Without SPI, the way to initialize a partial SDK would be to use `DefaultOpenTelemetry`.
+
+```java
+@Bean
+public OpenTelemetry openTelemetry() {
+  return DefaultOpenTelemetry.builder()
```

Comment:
So what is the essential difference between this case and the "full" `OpenTelemetrySdkBuilder` above?

Reply:
The SDK builder returns types that expose forceFlush and shutdown as required by the spec. Hence me filing issues to try to get rid of those :-)